hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/rmapp/attempt/RMAppAttemptMetrics.java

```diff
@@ -63,13 +69,18 @@
       new int[NodeType.values().length][NodeType.values().length];
   private volatile int totalAllocatedContainers;
 
+  private ConcurrentHashMap<Long, Long> allocationGuaranteedLatencies = null;
```

Comment:
Can you leave it without the null setting and make it final so the constructor sets it and that's it?

Reply:
Fixed.